OrbitClientServices/ProcessManager.cpp

```diff
@@ -124,11 +126,16 @@ void ProcessManagerImpl::WorkerFunction() {
       continue;
     }
 
-    absl::MutexLock callback_lock(&mutex_);
-    const auto& processes = result.value();
-    process_list_.assign(processes.begin(), processes.end());
-    if (process_list_update_listener_) {
-      process_list_update_listener_(this);
+    {
+      absl::MutexLock process_lock(&process_list_mutex_);
+      const auto& processes = result.value();
+      process_list_.assign(processes.begin(), processes.end());
```

Comment:
Does anyone use process_list_ ?

Reply:
Hm, good point. Actually, no. It's returned by GetProcessList(), but after this change, this isn't used at all. But I'm not sure if it will be needed with @antonrohr 's changes?

The current fix is in his change as a cherrypick because it was causing a crash, so I don't want to change this all too much before rebasing, but we could also not merge this fix and remove the whole GetProcessList() (and probably the mutex altogether...) afterwards.